### PR TITLE
FIx sizevars typechecking and add to mypy file

### PR DESCRIPTION
**Pull request by Chillee**
Mostly the same as this (https://github.com/pytorch/torchdynamo/pull/880) but without ghstack lol.

### CODE REVIEW

File: torchinductor/sizevars.py [hunk 1]
```diff
@@ -314,7 +320,8 @@ def strideof(name):
             code.writeline(f"{name}_stride = {name}.stride()")
             return f"{name}_stride"
 
-        needed = set(map(str, self.var_to_val.keys())) - set(self.replacements.keys())
+        needed_vars = set(self.var_to_val.keys()) - set(self.replacements.keys())
```
Comment (Chillee) [inline torchinductor/sizevars.py:323]: @jansel I think this was previously just a bug? `self.replacements` is a dict from `sympy.Symbol` to `sympy.Expr`. And here you're deduplicating against a set with strings.

Comment (jansel) [reply torchinductor/sizevars.py:323]: I think you are correct

**Review by jansel — APPROVED** [review]

File: torchinductor/sizevars.py (hunk 1 above)
Comment (Chillee) [reply torchinductor/sizevars.py:323]: Seems like fixing this bug causes some issues in `test/test_torchinductor.py::GpuTests::test_triton_conv_cuda`.

I get a `NameError: name 's0' is not defined` (cc: @pyjhzwh )

Comment (Chillee) [reply torchinductor/sizevars.py:323]: Gonna revert to the old behavior for now and add a note.

**Pull request merged**